armory/utils/metrics.py

```diff
@@ -431,6 +431,192 @@ def _check_object_detection_input(y, y_pred):
+        false_positives = np.zeros(len(class_predicted_boxes))
+
+        # Iterating over all predicted boxes of class_id
+        for pred_idx, pred_box in enumerate(class_predicted_boxes):
+            # Only compare gt boxes from the same image as the predicted box
+            gt_boxes_from_same_img = [
+                gt_box
+                for gt_box in class_gt_boxes
+                if gt_box["img_idx"] == pred_box["img_idx"]
+            ]
+
+            # If there are no gt boxes in the predicted box's image that have the predicted class
+            if len(gt_boxes_from_same_img) == 0:
+                false_positives[pred_idx] = 1
+                continue
+
+            # Iterate over all gt boxes (of class_id) from the same image as the predicted box, d
+            # etermining which gt box has the highest iou with the predicted box
+            highest_iou = 0
+            for gt_idx, gt_box in enumerate(gt_boxes_from_same_img):
+                iou = _intersection_over_union(pred_box["box"], gt_box["box"])
+                if iou >= highest_iou:
+                    highest_iou = iou
+                    highest_iou_gt_idx = gt_idx
+
+            if highest_iou > IOU_THRESHOLD:
+                # If the gt box has not yet been covered
+                if num_gt_boxes_per_img[pred_box["img_idx"]][highest_iou_gt_idx] == 0:
+                    true_positives[pred_idx] = 1
+
+                    # Record that we've now covered this gt box. Any subsequent
+                    # pred boxes that overlap with it are considered false positives
+                    num_gt_boxes_per_img[pred_box["img_idx"]][highest_iou_gt_idx] = 1
+                else:
+                    # This gt box was already covered previously (i.e a different predicted
+                    # box was deemed a true positive after overlapping with this gt box)
+                    false_positives[pred_idx] = 1
+            else:
+                false_positives[pred_idx] = 1
+
+        # Cumulative sums of false/true positives across all predictions which were sorted by
+        # descending confidence
+        tp_cumulative_sum = np.cumsum(true_positives)
+        fp_cumulative_sum = np.cumsum(false_positives)
+
+        # Total number of gt boxes with a label of class_id
+        total_gt_boxes = len(class_gt_boxes)
+
+        recalls = tp_cumulative_sum / (total_gt_boxes + 1e-6)
+        precisions = tp_cumulative_sum / (tp_cumulative_sum + fp_cumulative_sum + 1e-6)
+
+        interpolated_precisions = np.zeros(len(RECALL_POINTS))
+        # Interpolate the precision at each recall level by taking the max precision for which
+        # the corresponding recall exceeds the recall point
+        # See http://citeseerx.ist.psu.edu/viewdoc/download?doi=10.1.1.157.5766&rep=rep1&type=pdf
+        for i, recall_point in enumerate(RECALL_POINTS):
+            precisions_points = precisions[np.where(recalls > recall_point)]
```

Comment:
Per the link, I think this should be `recalls >= recall_point`

Reply:
agreed, will update